### ISSUE TITLE
builder.Services.AddFeatureManagement() with configuration section does not work since v3.0.0

### ISSUE THREAD

**Issue by FredericVaugeoisFlo** [microsoft/FeatureManagement-Dotnet#308]
I know that there was a breaking change in v3.0.0 created by https://github.com/microsoft/FeatureManagement-Dotnet/pull/261, but I feel like that should not affect the use case of targeting the configuration section to use in the appsettings.json file.

For example, with  this configuration file:

```json
{
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft.AspNetCore": "Warning"
    }
  },
  "FeatureFlags": {
    "PushFeatureEnabled": true,
    "PutFeatureEnabled": true
  }
}
```

and this program.cs line:

```csharp
builder.Services.AddFeatureManagement(builder.Configuration.GetSection("FeatureFlags"));

var app = builder.Build();
```

the feature manager does not load our feature flags like it did in 2.6.0. If we're not allowed to set our own configuration section, it should no be a possible parameter or at least raise a deprecated warning

**Comment by zhiyuanliang-ms** [comment]
https://learn.microsoft.com/en-us/azure/azure-app-configuration/use-feature-flags-dotnet-core?tabs=core6x
![image](https://github.com/microsoft/FeatureManagement-Dotnet/assets/141655842/48679c1e-8a0b-4ef2-a7a5-5fb588b8b909)
In our public document, we mentioned this usage.

**Comment by jimmyca15** [comment]
Agree, we should preserve the behavior for this override. Thanks for raising the issue.

**Comment by FredericVaugeoisFlo** [comment]
Great thank you!

**Comment by TobiasGPersson** [comment]
Will this feature be patched to work for .NET 8 again? It would be relevant to name the configuration-section to a custom name instead of relying on only "FeatureManagement".

**Comment by jimmyca15** [comment]
> Will this feature be patched to work for .NET 8 again? 

I don't expect any specific issue for .NET 8. All versions of .NET should behave the same.

> It would be relevant to name the configuration-section to a custom name instead of relying on only "FeatureManagement".

Is this a request for that capability? That should be doable once we address the issue using the pattern in the initial issue description. `services.AddFeatureManagement(configuration.GetSection("MyCustomFlagsSection"));`

**Comment by zhiyuanliang-ms** [comment]
Fixed in release 3.1.0
